clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -8522,6 +8522,22 @@ bool Sema::CheckCountedByAttr(Scope *S, const FieldDecl *FD) {
     }
   }
 
+  // We don't support 'counted_by' on flexible array members in substructures.
+  const DeclContext *DC = FD->getParent();
+  while (const auto *RD = dyn_cast<RecordDecl>(DC)) {
+    if (!RD->isAnonymousStructOrUnion() ||
+        !isa<RecordDecl>(RD->getLexicalParent()))
+      break;
+    DC = RD->getLexicalParent();
+  }
+
+  if (DC != FD->getDeclContext()->getOuterLexicalRecordContext()) {
```

Comment:
We need to figure out this thing about `getOuterLexicalRecordContext()` before we can go forward here.  It sounds like you might be concerned about the established behavior of some existing attribute that isn't the `-fbounds-safety` `__counted_by`, but which people are trying to merge with `__counted_by` on general principle?

Reply:
I added this to ensure that we're not dealing with FAMs in sub-structures. I need to work with @rapidsna's work to make sure that what they do is what this attribute will do. It's not really a perfect check, but I removed the code in CodeGen that handles FAMs in substructs. So I think we can table this discussion if this patch goes on.